### PR TITLE
fixed subjectSplitter and added test for subjects unassigned to folds

### PR DESCRIPTION
**Pull request by egillax**
There was an issue with the subjectSplitter function. 

It uses the R function round() to get indices for the train and test set. But when the expression inside the round ends in a .5 this function rounds to the nearest even number. For example if it was 20.5 the rounding would result in 20. So the test set would for example go from 1:20. But then for the training set we add 1 to the index, so it's 21.5 and then the rounding gives 22. So the training set is from 22 and upwards. So subject 21 is left out.  

I first added a test to test this situation and indeed it failed with the current code. Then I modified the code so it always rounds up when the expression ends in .5. Then the test passed.

I ran the tests for all the data-splitting functions and they all pass. But I'm struggling to run all of the tests for the package, is there guidance somewhere on how to do that?

Egill

### CODE REVIEW

**Comment by jreps** [comment]
Great find.  I always wondered why the split sometime did weird things.  I can't believe that round() doesn't actually round the numbers.  Thanks for fixing this.

For the testing:  locally you can use R studio (you need rtools installed in windows I think and Xcode in Mac) - in r studio on the top right you'll see 'Environment', 'History', 'Connections', 'Build' tabs - click 'Build' and then in the tab options below click 'more' then select 'Test Package' - this will run all the tests.

The tests via GitHub Actions should also run when you use GitHub from now on (I had to authorize it).
